internal/restorer/restorer_test.go

```diff
@@ -276,6 +286,9 @@ func TestRestorer(t *testing.T) {
 			tempdir, cleanup := rtest.TempDir(t)
 			defer cleanup()
 
+			// make sure we're creating a new subdir of the tempdir
+			tempdir = filepath.Join(tempdir, "target")
```

Comment:
Sigh. This is broken on master already. Regression I introduced in #1772. Sorry about that. The fix is to eagerly create `dst` directory in `restorer.RestoreTo`

```go
	err = fs.MkdirAll(dst, 0777) // umask takes care of dir permissions
	if err != nil {
		return errors.Wrap(err, "MkdirAll")
	}
```

Let me know if you'd like separate PR to address the regression on master.

Reply:
Cool, thanks for looking into the issue! I'll do a quick PR to fix it.

Reply:
Will be fixed in #1885

Reply:
@ifedorenko Now that #1885 is merged, does that require a rebase on this branch? Would be exciting to be able to see this PR make it in. :)

Reply:
I assumed @fd0 was working on this. I can rebase this PR some time next week, if that helps.

Reply:
I'm working on it, and I'll rebase this shortly.